lib/backend/backend.go

```diff
@@ -38,6 +39,10 @@ const (
 	Forever time.Duration = 0
 )
 
+// ErrIncorrectRevision is returned from conditional operations when revisions
+// do not match the expected value.
+var ErrIncorrectRevision = &trace.CompareFailedError{Message: "resource revision does not match, it may have been concurrently created|modified|deleted; please work from the latest state, or use --force to overwrite"}
```

Comment:
Are we expecting callers to `errors.Is(err, backend.ErrIncorrectRevision)`, as opposed to `trace.IsCompareFailed`? And would that even work just for this specific instance of the error?